File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
#include <string>

class RobotNewsStationNode : public rclcpp::Node
{
public:
    RobotNewsStationNode() : Node("robot_news_station")
    {
        __publisher = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        __timer = this->create_wall_timer(std::chrono::milliseconds(500),
                                          std::bind(&RobotNewsStationNode::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "C++ Robot News Station node has started!");
    }

private:
    rclcpp::TimerBase::SharedPtr __timer;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr __publisher;

    void publishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = "Hello from C++ Robot News Station publisher!";
        __publisher->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); // Cpp nodes are created as shared pointer
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import (
    String,
)  # need to add this dependency in package.xml too


class RobotNewsStationNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")

        self.publisher = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(1, self.publish_news)
        self.get_logger().info("Python Robot News Station node has started!")

    def publish_news(self):
        msg = String()
        msg.data = "Hello from Python Robot News Station publisher!"
        self.publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)  # initialize ROS
    node = RobotNewsStationNode()  # create node
    rclpy.spin(node)  # allows the node to run indefintely
    rclpy.shutdown()  # shutdown node


if __name__ == "__main__":
    main()
